File: Python/problem001.py
#! /usr/bin/env python3
"""
Problem 1 - Project Euler
http://projecteuler.net/index.php?section=problems&id=001
"""


def summul(n, x):
    return int(x * (n // x) * (n // x + 1) / 2)


if __name__ == "__main__":
    N = 1000 - 1
    print(summul(N, 3) + summul(N, 5) - summul(N, 3 * 5))

File: Scala/problem001.scala
// Problem 1 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=1

object Problem001 {
  def summul(n: Int, x: Int): Int = {
    val count = n / x
    x * (count * (1 + count) / 2)
  }

  def main(args: Array[String]) : Unit = {
    println(summul(999, 3) + summul(999, 5) - summul(999, 15))
  }
}
